geomesa-raster/src/main/scala/org/locationtech/geomesa/raster/data/AccumuloRasterStore.scala

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright 2014 Commonwealth Computer Research, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.locationtech.geomesa.raster.data
+
+import java.io.Serializable
+import java.util.Map.Entry
+import java.util.concurrent.{Callable, TimeUnit}
+import java.util.{Map => JMap}
+
+import com.google.common.cache.CacheBuilder
+import com.google.common.collect.ImmutableSetMultimap
+import com.typesafe.scalalogging.slf4j.Logging
+import org.apache.accumulo.core.client.{BatchWriterConfig, Connector, TableExistsException}
+import org.apache.accumulo.core.data.{Key, Mutation, Range, Value}
+import org.apache.accumulo.core.security.TablePermission
+import org.geotools.coverage.grid.GridEnvelope2D
+import org.joda.time.DateTime
+import org.locationtech.geomesa.accumulo.index.Strategy._
+import org.locationtech.geomesa.accumulo.iterators.BBOXCombiner._
+import org.locationtech.geomesa.accumulo.stats.{RasterQueryStat, RasterQueryStatTransform, StatWriter}
+import org.locationtech.geomesa.accumulo.util.{SelfClosingBatchScanner, SelfClosingScanner}
+import org.locationtech.geomesa.raster._
+import org.locationtech.geomesa.raster.index.RasterIndexSchema
+import org.locationtech.geomesa.raster.util.RasterUtils
+import org.locationtech.geomesa.security.AuthorizationsProvider
+import org.locationtech.geomesa.utils.geohash.BoundingBox
+import org.locationtech.geomesa.utils.stats.{MethodProfiling, NoOpTimings, Timings, TimingsImpl}
+
+import scala.collection.JavaConversions._
+
+class AccumuloRasterStore(val connector: Connector,
```

Comment:
[minor] [for the future] I wonder if there is a better way/place to set the default values for writeMemory, writeThreads and numQThreads. I see this approach is also taken in the AccumuloDataStore. @jnh5y: would it make sense to consolidate the defaults for memory and threading parameters, etc. into a trait or a config? With some comments regarding why the values were chosen? The magic numbers are a bit suspicious, although I'll admit that being at the top of the class constructor they aren't exactly buried.